subsys/usb/class/cdc_acm.c

```diff
@@ -390,16 +405,20 @@ static void cdc_acm_do_cb(struct cdc_acm_dev_data_t *dev_data,
 		break;
 	case USB_DC_RESET:
 		LOG_DBG("USB device reset detected");
+		cdc_acm_reset_port(dev_data);
 		break;
 	case USB_DC_CONNECTED:
 		LOG_DBG("USB device connected");
 		break;
 	case USB_DC_CONFIGURED:
 		dev_data->tx_ready = true;
+		dev_data->tx_irq_ena = true;
```

Comment:
you changed logic here, is this needed?

Reply:
Yes, because interrupts are disabled during USB reset and if they're not enabled, communication doesn't work.
In cdc_acm example enabling rx IRQ is done in initialization:
```
/* Enable rx interrupts */
uart_irq_rx_enable(dev);
```
So if there is a USB reset after that, the stack should re-enable them.

What could be considered is moving interrupts enabling and disabling to USB stack completely. Is there a use case for having them enabled and disabled inside application?